pkg/service/cluster.go

```diff
@@ -596,6 +596,13 @@ func (s *clusterService) Update(ctx context.Context, cluster *infrav3.Cluster) (
 		h.OnChange(ev)
 	}*/
 
+	if id, err := uuid.Parse(projectName); err == nil {
```

Comment:
This looks good. This safeguards us irrespective of whether project name or id is present in the `projectName` variable. Just curious as to when/why are we passing project id instead of name.